docs/docs/use-cases/flux.md

```diff
@@ -0,0 +1,186 @@
+
+Flux is a tool that keeps Kubernetes clusters synchronized with sources
+of configuration (such as Git repositories) and automates updates to
+configuration when new code is deployed.
+
+This section shows how to add Keptn to an existing Flux use case that runs
+[pre and post-deployment jobs with Flux](https://fluxcd.io/flux/use-cases/running-jobs/)
+to make it simpler and more straight-forward.
+
+## High-level structure of the Git repository
+
+Since Flux uses a GitOps approach for continuous delivery, the Git
+repository structure for our use case looks like the following:
+
+```markdown
+├── apps
+│   └── dev
+│       ├── podtato-head.yaml
+│       └── kustomize.yaml
+└── clusters
+    └── dev
+        ├── flux-system
+        │   ├── gotk-components.yaml
+        │   ├── gotk-sync.yaml
+        │   └── kustomize.yaml
+        ├── podtato-head-source.yaml
+        └── podtato-head-kustomization.yaml
+```
+
+The `apps` directory contains application manifests, that will be deployed.
+The `clusters` directory contains Flux configuration manifests and custom
+resources, that are needed for the delivery.
+`apps` and `clusters` directories can live in two separate repositories,
+but for simplicity of this excercise, we will keep them in a single one.
+
+## Set up your environment
+
+Before we start, you need to install Flux CLI and Keptn.
+You can find the installation instructions for Keptn [here](../installation/index.md)
+and for Flux [here](https://fluxcd.io/flux/installation/).
+
+After successfully installing Flux CLI and Keptn, you need to
+retrieve your Git repository credentials.
+You can use any available Git providers, but be sure to store your `token`
+for later usage.
+For simplicity, we will use GitHub.
+
+You also need to install Flux in your cluster.
+This requires
+[bootstrapping the git repository](https://fluxcd.io/flux/installation/bootstrap/)
+to set up all necessary Flux structures in the repository.
+Use the following command to do this:
+
+```shell
+GITHUB_USER=<user-name> GITHUB_TOKEN=<token> \
+flux bootstrap github \
+  --owner=$GITHUB_USER \
+  --repository=podtato-head \
+  --branch=main \
+  --path=./clusters/dev \
+  --personal
+```
+
+The bootstrap command above does the following:
+
+* Creates a Git repository `podtato-head` on your GitHub account.
+* Adds Flux component manifests to the repository -
+creates `./clusters/dev/flux-system/*` structure.
+* Deploys Flux components to your Kubernetes Cluster.
+* Configures Flux components to track the path `./clusters/dev/` in the repository.
+
+## Creating the application
+
+Now it's time to add the application together with pre- and
+post-deployment checks to the repository.
+
+First, clone the `podtato-head` repository to your local machine.
+
+Add the following manifests that represent the application into the  `podtato-head.yaml` file
+and store it in the `./apps/dev/` directory of your repository:
+
+```yaml
+{% include "./assets/flux/app.yaml" %}
+```
+
```

Comment:
Would it be useful to say a little more about this manifest.  For example:

This manifest defines the following Kubernetes and Keptn resources:

- Kubernetes
    - Namespace -- enables Keptn for this namespace
    = Deployment - defines labels and annotations that integrate Keptn with the deployment
- Keptn
     - KeptnAppContext -- lists the pre- and post-deployment tasks to run.  In this case, we have just one pre-deployment task -- hello-task -- to run
     - KeptnTaskDefinition -- defines the executable code for the `hello-task` task.

Something like that.  If you like the idea, I'll code up the links to the k8s and Keptn doc for each resource and work a little harder on the descriptions.

Reply:
Is it actually needed? The resources are completely standard and explained in other guides/ places in our docs. I would rather keep it simple and straight-forward.

Reply:
Yes, it's a question of whether we force the reader to go digging for the information or give them a link so they easily find the relevant info.   But this is you call.  You can just resolve this issue if you don't want the list-with-links -- that's why I didn't bother to do the coding ;-)

Reply:
I would rather refer to the guide on deploying a keptn app to help user here, instead of explaining in details the basics... IMO flux is not a beginner tool so the risk is making the page unreadable for those who know what they are doing...

Reply:
It looks like Ondrej accepted my suggestion to include a link to the CRD Ref pages and "Basic annotations".  I'm marking this as resolved.